OpenRA.Mods.Common/Widgets/ObserverArmyIconsWidget.cs

```diff
@@ -149,51 +149,40 @@ public override Widget Clone()
 			return new ObserverArmyIconsWidget(this);
 		}
 
-		public override void MouseEntered()
+		public override void Tick()
 		{
 			if (TooltipContainer == null)
 				return;
 
-			foreach (var armyIcon in armyIcons)
+			if (Ui.MouseOverWidget != this)
```

Comment:
I finally worked out why the tooltips weren't showing when the mouse was to the right of the last icon.
Replacing the `Bounds.Width` assignment in `Draw` with the following will resolve the issue:
```csharp
var newWidth = queueCol * (IconWidth + IconSpacing);
if (newWidth != Bounds.Width)
{
	var wasInBounds = EventBounds.Contains(Viewport.LastMousePos);
	Bounds.Width = newWidth;
	var isInBounds = EventBounds.Contains(Viewport.LastMousePos);

	// HACK: Ui.MouseOverWidget is normally only updated when the mouse moves
	// Call ResetTooltips to force a fake mouse movement so the checks in Tick will work properly
	if (wasInBounds != isInBounds)
		Game.RunAfterTick(Ui.ResetTooltips);
}
```

Reply:
Did not retest it after changing this check from RenderBounds.Contain() to Ui.MouseOverWidget != this. So change back to the previous code would be another solution.

Reply:
Yeah, although I would prefer the fix here because its more general (other code may be using the wrong value of `Ui.MouseOverWidget`) and avoids calling `EventBounds.Contains()` every tick.